docs/upgrading_wagtail_v2/index.rst

```diff
@@ -0,0 +1,158 @@
+==============================
```

Comment:
1. This document should also be linked from `/docs/releases/2.0.rst` I guess.
1. Is `/docs/upgrading_wagtail_v2/` the right place? We already have `/docs/releases/upgrading.rst` so maybe it should leave alongside to it?